File: icommerce/src/main/java/com/icommerce/modelo/Producto.java
package com.icommerce.modelo;


import java.util.List;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.Data;

/**
 *
 * @author Miguel
 */
@Data
@Entity
@Table(name = "producto")
public class Producto{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    
    @Column(name = "nombre")
    private String nombre;
    
    
    @Column(name = "precio")
    private float precio;
    
    
    @Column(name = "imagen")
    private String imagen;
    
    @Column(name = "activo")
    private Boolean activo;
    
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "producto")
    private List<MenuProducto> menus;
    
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "producto")
    private List<PedidoProducto> pedidos;


    
}

File: icommerceApp/src/app/modelo/Producto.ts
import { Pedido } from "./Pedido";

export class Producto{
    constructor(public id:number,
        public nombre:string,
        public precio:number,
        public imagen:string,
        public activo:boolean){}

    public static createFromJsonObject(jsonObject:any):Producto
    {
        let producto:Producto = new Producto(jsonObject['id'],
        jsonObject['nombre'],
        jsonObject['precio'],
        jsonObject['imagen'],
        jsonObject['activo']);
        return producto;
    }
}